packages/rate-limit-controller/src/RateLimitController.ts

```diff
@@ -164,15 +164,15 @@ export class RateLimitController<
     }
     this.recordRequest(type, origin);
 
-    const implementation = this.implementations[type].method;
+    const implementation = this.implementations[type].method as (
+      ...args: Parameters<RateLimitedApis[ApiType]['method']>
+    ) => ReturnType<RateLimitedApis[ApiType]['method']>;
 
     if (!implementation) {
       throw new Error('Invalid api type');
     }
 
-    return implementation(...args) as ReturnType<
-      RateLimitedApis[ApiType]['method']
-    >;
+    return implementation(...args);
```

Comment:
@MetaMask/snaps-devs This change fixes the following regression:

```ts
ERROR: src/RateLimitController.ts:173:27 - error TS2345: Argument of type 'Parameters<RateLimitedApis[ApiType]["method"]>' is not assignable to parameter of type 'never'.
  Type 'unknown[]' is not assignable to type 'never'.
    Type 'Parameters<RateLimitedApis[keyof RateLimitedApis]["method"]>' is not assignable to type 'never'.
      Type 'unknown[]' is not assignable to type 'never'.
        Type 'Parameters<RateLimitedApis[string | number | symbol]["method"]>' is not assignable to type 'never'.
          Type 'unknown[]' is not assignable to type 'never'.
            Type 'Parameters<RateLimitedApis[string]["method"]> | Parameters<RateLimitedApis[number]["method"]> | Parameters<RateLimitedApis[symbol]["method"]>' is not assignable to type 'never'.
              Type 'Parameters<RateLimitedApis[string]["method"]>' is not assignable to type 'never'.
                Type 'unknown[]' is not assignable to type 'never'.
                  Type 'never[]' is not assignable to type 'never'.

173     return implementation(...args) as ReturnType<
```

Both before and after the change, the as assertion narrows the type of `implementation`, which is currently typed to the universal supertype (i.e. type constraint) of all `RateLimitedApi` `method` types (i.e. `ActionConstraint`) because there's no `ApiType` information to narrow it.

The only difference is that the type assertion now narrows the parameters (`...args`) of `implementation`, in addition to its return type.